pkg/cmd/account/aws/create/create_test.go

```diff
@@ -0,0 +1,146 @@
+	"github.com/stretchr/testify/assert"
+)
+
+var serverUrl, _ = url.Parse("http://server")
+
+const placeholderApiKey = "API-XXXXXXXXXXXXXXXXXXXXXXXXXXXXX"
+
+var rootResource = testutil.NewRootResource()
+
+func TestAWSAccountCreatePropmtMissing(t *testing.T) {
+	const spaceID = "Space-1"
+	const envID = "Env-1"
+	_ = fixtures.NewSpace(spaceID, "testspace")
+	env := fixtures.NewEnvironment(spaceID, envID, "testenv")
+	api, qa := testutil.NewMockServerAndAsker()
+	out := &bytes.Buffer{}
+
+	opts := &create.CreateOptions{
+		CreateFlags: create.NewCreateFlags(),
+	}
+
+	errReceiver := testutil.GoBegin(func() error {
+		defer testutil.Close(api, qa)
+		octopus, _ := octopusApiClient.NewClient(testutil.NewMockHttpClientWithTransport(api), serverUrl, placeholderApiKey, "")
+		opts.Ask = qa.AsAsker()
+		opts.Octopus = octopus
+		opts.Writer = out
+		return create.PromptMissing(opts)
+	})
+
+	api.ExpectRequest(t, "GET", "/api").RespondWith(rootResource)
+
+	_ = qa.ExpectQuestion(t, &survey.Input{
+		Message: "Name",
+		Help:    "A short, memorable, unique name for this account.",
+	}).AnswerWith("TestAccount")
+
+	_ = qa.ExpectQuestion(t, &surveyext.OctoEditor{
+		Editor: &survey.Editor{
+			Message:  "Description",
+			Help:     "A summary explaining the use of the account to other users.",
+			FileName: "*.md",
+		},
+		Optional: true,
+	}).AnswerWith("test 123")
+
+	_ = qa.ExpectQuestion(t, &survey.Input{
+		Message: "Access Key",
+		Help:    "The AWS access key to use when authenticating against Amazon Web Services.",
+	}).AnswerWith("testaccesskey123")
+
+	_ = qa.ExpectQuestion(t, &survey.Password{
+		Message: "Secret Key",
+		Help:    "The AWS secret key to use when authenticating against Amazon Web Services.",
+	}).AnswerWith("testpassword124")
+
+	api.ExpectRequest(t, "GET", "/api/Spaces-1/environments").RespondWith(resources.Resources[*environments.Environment]{
+		Items: []*environments.Environment{env}},
+	)
+
+	_ = qa.ExpectQuestion(t, &survey.MultiSelect{
+		Message: "Choose the environments that are allowed to use this account.\nIf nothing is selected, the account can be used for deployments to any environment.",
+		Options: []string{"testenv"},
+	}).AnswerWith([]string{"testenv"})
+
+	err := <-errReceiver
+	assert.Nil(t, err)
+
+	assert.Equal(t, []string{envID}, opts.Environments.Value)
+	assert.Equal(t, "testpassword124", opts.SecretKey.Value)
+	assert.Equal(t, "testaccesskey123", opts.AccessKey.Value)
+	assert.Equal(t, "test 123", opts.Description.Value)
+	assert.Equal(t, "TestAccount", opts.Name.Value)
+}
+
+func TestAWSAccountCreateNoPrompt(t *testing.T) {
+	const spaceID = "Space-1"
+	const envID = "Env-1"
+	_ = fixtures.NewSpace(spaceID, "testspace")
+	_ = fixtures.NewEnvironment(spaceID, envID, "testenv")
+	api, qa := testutil.NewMockServerAndAsker()
+	out := &bytes.Buffer{}
+
+	opts := &create.CreateOptions{
+		CreateFlags: create.NewCreateFlags(),
+	}
+	opts.Space = spaceID
```

Comment:
As above, trivial, but maybe nicer if these are assigned inside the `{ }` for CreateOptiosn